tests/test-private-upload.ts

```diff
@@ -0,0 +1,128 @@
+import { GetObjectCommand, S3Client, S3ClientConfig } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { test, expect } from '@jest/globals';
+import { MiddlewareType } from '@smithy/types';
+
+import { handler } from '../src/lambda-handler';
+import { Args } from '../src/lib';
+
+/**
+ * Presign a URL for a given key.
+ * @param key
+ * @returns {Promise<string>}
+ */
+async function getPresignedUrlParams( key: string ) : Promise<Args> {
+	const client = new S3Client(
+		{
+			region: process.env.S3_REGION,
+			bucket: process.env.S3_BUCKET,
+		} as S3ClientConfig
+	);
+	const command = new GetObjectCommand( {
+		Bucket: process.env.S3_BUCKET,
+		Key: key,
+	} );
+
+	/**
+	 * Middleware to remove the x-id query string param form the GetObject call.
+	 */
+	const middleware: MiddlewareType<any, any> = ( next: any, context: any ) => async ( args: any ) => {
+		const { request } = args;
+		delete request.query['x-id'];
+		return next( args );
+	};
+
+	client.middlewareStack.addRelativeTo( middleware, {
+		name: 'tests',
+		relation: 'before',
+		toMiddleware: 'awsAuthMiddleware',
+		override: true,
+	} );
+
+	const presignedUrl = new URL( await getSignedUrl( client, command, {
+		expiresIn: 60,
+	} ) );
+	// console.log( presignedUrl );
+	let queryStringParameters: Args = {};
+	presignedUrl.searchParams.forEach( ( value, key ) => {
+		queryStringParameters[ key as keyof Args ] = value;
+	} );
+
+	return queryStringParameters;
```

Comment:
Maybe: `Object.fromEntries( presignedUrl.searchParams )` ?

Reply:
Ah that's the function I was looking for I think!